frontend/openchat-agent/src/services/openchatAgent.ts

```diff
@@ -851,6 +869,129 @@ export class OpenChatAgent extends EventTarget {
             });
     }
 
+    async getInitialStateV2(): Promise<ChatStateFull> {
+        const cached = await getCachedChatsV2(this.db, this.principal);
+        if (cached !== undefined) {
+            return await this.getUpdatesV2(cached);
+        }
+
+        let state: ChatStateFull;
+        const userResponse = await this.userClient.getInitialStateV2();
+        if (userResponse.cacheTimestamp === undefined) {
+            const groupPromises = userResponse.groupChatsAdded.map((g) => this.getGroupClient(g.chatId).summary());
+            const groupChats = (await Promise.all(groupPromises)).filter(isSuccessfulGroupSummaryResponse);
```

Comment:
There are potential problems here with all uses of `Promise.all`. If any of the promises reject, the whole `Promise.all` will reject. Because you are using async/await you will get an exception. We probably need to do _something_ about this. When `Promise.all` rejects (or throws in the async/await case) the error will itself have an `errors` propery which will be an array of errors for the rejected promises. We need to at least handle this otherwise it could blow up. Unless we decide that in that case it _should_ blow up.